src/components/HomeSlider.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from "react";
+
+const HomeSlider = ({ banners }) => {
+  const [current, setCurrent] = useState(0);
+  const length = banners.length;
+
+  const nextSlide = () => {
+    setCurrent(current === length - 1 ? 0 : current + 1);
+  };
+
+  if (!Array.isArray(banners) || banners.length <= 0) {
+    return null;
+  }
+
+  setTimeout(() => {
+    var slider = document.getElementById("home-carrousel");
+
+    if (!slider) {
+      return;
+    }
+    nextSlide();
+  }, 20000);
+
+  return (
+    <section className="Slider" id="home-carrousel">
+      {banners.map((slide, index) => {
+        return (
+          <div
+            className={index === current ? "slide active" : "slide"}
+            key={index}
+          >
+            {index === current && (
+              <div>
+                <img
+                  src={slide.data.main_image.url}
+                  alt={"banner " + (index + 1)}
+                  className="Banner-image"
+                />
+                <div className="Banner-title">
+                  {slide.data.title}
+                </div>
+                <div className="Banner-description">
+                  {slide.data.description[0].text}
```

Comment:
Let's keep the json structure implementation detail out of the `HomeSlider` component. We can, for example, receive in the props an array of banners that have the properties `description`, `imageSource` and `title` and let another component or hook extract these values from the json